core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -430,8 +476,50 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
     newIterator.foreach(addIfAbsent)
     oldIterator.foreach(addIfAbsent)
+    mergedApps
+  }
+
+
+  /**
+   * Scan through all the application attempts, if any have changed those attempts
+   * will be updated with the new file sizes. No attempt to replay the application
+   * is made; this is a low cost operation.
+   */
+  private[history] def updateAttemptFileSizes(): Unit = {
+    val now = System.currentTimeMillis();
+    val newAttempts: Iterable[FsApplicationAttemptInfo] = applications
+        .filter( e => !e._2.completed)
+        .flatMap { e =>
+          // build list of (false, attempt) or (true, attempt') values
```

Comment:
noted, will fix. The existing scanner finds new files; this just looks for changes